nalgebra-sparse/src/io/matrix_market.rs

```diff
@@ -0,0 +1,1293 @@
+//! Implementation of matrix market io code.
+//!
+//! See the [website](https://math.nist.gov/MatrixMarket/formats.html) or the [paper](https://www.researchgate.net/publication/2630533_The_Matrix_Market_Exchange_Formats_Initial_Design) for more details about matrix market.
+use crate::coo::CooMatrix;
+use crate::SparseFormatError;
+use crate::SparseFormatErrorKind;
+use nalgebra::base::Scalar;
+use nalgebra::Complex;
+use pest::iterators::Pairs;
+use pest::Parser;
+use std::cmp::PartialEq;
+use std::convert::Infallible;
+use std::convert::TryFrom;
+use std::fmt;
+use std::fmt::Formatter;
+use std::fs;
+use std::num::ParseIntError;
+use std::num::TryFromIntError;
+use std::path::Path;
+use std::str::FromStr;
+
+/// A description of the error that occurred during importing a matrix from a matrix market format data.
+#[derive(Debug)]
+pub struct MatrixMarketError {
+    error_kind: MatrixMarketErrorKind,
+    message: String,
+}
+
+/// Errors produced by functions that expect well-formed matrix market format data.
+/// > _NOTE:_ Since the matrix market design didn't mention if multiple sparse entries with the same coordiantes are allowed or not, so, it's allowed here.
+#[non_exhaustive]
+#[derive(Copy, Clone, Debug, PartialEq)]
```

Comment:
I'd perhaps like to avoid the `Copy` impl for now, because I'm not sure if we really want to keep it `Copy` in the future.

Reply:
it's actually because both `OperationErrorKind` and `SparseFormatErrorKind` implement the copy trait, so I kind of copied that

Reply:
actually, I tried remove the `Copy` impl, it gives compiling error ` type could implement `Copy`; consider adding `impl Copy``, because of `#![deny(missing_copy_implementations)]`